src/ride/views.py

```diff
@@ -1 +1,45 @@
-# Create your views here.
+import json
+
+from api import settings as api_settings
+from api.utils import failure_response
+from api.utils import success_response
+from rest_framework import generics
+from rest_framework import status
+
+from .controllers.create_ride_controller import CreateRideController
+from .models import Ride
+from .serializers import RideSerializer
+
+
+class AllRidesView(generics.GenericAPIView):
+    serializer_class = RideSerializer
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def get(self, request):
+        """Get all rides."""
+        return success_response(
+            self.serializer_class(Ride.objects.all(), many=True).data
+        )
+
+
+class RideView(generics.GenericAPIView):
+    serializer_class = RideSerializer
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def get(self, request):
+        """Get ride by id."""
+        id = request.query_params.get("id", None)
```

Comment:
is there a reason the ride id is a URL param instead of just being in the url? I feel like `/api/rides/<id>/` is cleaner (I'm biased bc we barely use URL params on pear)

Reply:
agree with noah, this one is essentially doing `/api/rides/id?<id>/`and `/api/rides/<id>/` is def cleaner

Reply:
good point! fixed :)